controllers/cluster_controller.go

```diff
@@ -178,6 +178,16 @@ func patchCluster(ctx context.Context, patchHelper *patch.Helper, cluster *clust
 
 // reconcile handles cluster reconciliation.
 func (r *ClusterReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx, "cluster", cluster.Name)
+
+	if cluster.Spec.Topology != nil {
+		if cluster.Spec.ControlPlaneRef == nil || cluster.Spec.InfrastructureRef == nil {
+			// TODO: add a condition to surface this scenario
+			log.Info("Waiting for the topology to be generated")
```

Comment:
Should we increase the verbosity of this log message?

Reply:
Looks like we have a lot of similar logs with the same verbosity. I would have assumed we don't get a lot of these logs as we should only hit this code path when a new cluster is created *and* hopefully the topology generation shouldn't take too long (or at least we shouldn't get a lot of updates on the cluster until it's generated)

Reply:
I think having the log error will help at this stage, we can eventually iterate later

Reply:
Will do this in a follow up PR, just saw your comment.